### ISSUE TITLE
(Nightly/msx) use of sprites in the glx library makes the display strange.

### ISSUE THREAD

**Issue by fu-sen** [z88dk/z88dk#1213]
When I run [examples/msx](https://github.com/z88dk/z88dk/tree/master/examples/msx) with the latest Nightly Build,
I get display strange on some sources:
![Dropbox_0042](https://user-images.githubusercontent.com/5434159/60435003-715a5c80-9c43-11e9-970b-5b8e89edf6be.jpg)
![Dropbox_0043](https://user-images.githubusercontent.com/5434159/60435009-74ede380-9c43-11e9-8c51-b5b9cbb67f46.jpg)
This seems to be showing SCREEN 3 (multi color mode) because it is a complete block and sprite.
Or it is completely black and no text is displayed.

It is the following source that gets these indications:
- ex4.c
- ex6.c
- ex7.c
- ex9.c
- ex11.c
- showlib3d.c

Many of them use sprites. It seems that 3D library will be added there.

**Comment by fu-sen** [comment]
I was searching for this cause. It is failing with **set_mangled_mode()**.
It is necessary to change VDP to SCREEN 2 after changing to SCREEN 1, which **requires a BIOS call.**
But you changed VDP directly to change the screen mode. It seems to be lost.

**Comment by fu-sen** [comment]
In the sample using "set_mode (mode_2);", some samples work, but ex9.c does not work.
So it seems that sprites also have problems.

**Comment by zx70** [comment]
your theory about the bios requirements could be partially verified with asimple test:  what happens if you add -lmsxbios to your compile command chain? 
please try also by running the makefile

**Comment by fu-sen** [comment]
You forgot past work. It is included in the Makefile:
[z88dk/examples/msx/Makefile](https://github.com/z88dk/z88dk/blob/master/examples/msx/Makefile):
```
EX8.COM: ex8.c
	$(CC) +msx -vn -subtype=msxdos -create-app -oex8 ex8.c -lmsxbios

ADVA.COM: ../console/adv_a.c
	$(CC) +msx -vn -subtype=msxdos ../console/adv_a.c -I../console -DCOMPUTE_SCORE -DPICS -DDISK -create-app -clib=ansi -oadva.bin

ex6.cas: ex6.c
	$(CC) $(CFLAGS) -oex6.bin ex6.c -lmsxbios

ex7.cas: ex7.c
	$(CC) $(CFLAGS) -oex7.bin ex7.c -lmsxbios


showlib3d.cas: showlib3d.c
	$(CC) $(CFLAGS) -oshowlib3d.bin showlib3d.c -lmsxbios

ex11.cas: ex11.c
	$(CC) $(CFLAGS) -oex11.bin ex11.c -lmsxbios
```

**Comment by zx70** [comment]
I forget many things :)
Is it still broken? I know there are still limits on the SVI ports but the MSX part looks ok to me.

**Comment by fu-sen** [comment]
Unfortunately, this issue is ongoing in the latest build (20191007).
Execution result of ex6.cas using blueMSX:
![ex6_0002](https://user-images.githubusercontent.com/5434159/66286847-30db4080-e90e-11e9-994b-faa1dad04a30.png)

**Comment by suborb** [comment]
I've just merged #1300 which puts the sprite address back to where they are the MSX. The sprite demos now seem to work. 

Additionally, msx_set_mangled_mode() is now restored.

If there's still issues, please reopen.

**Comment by fu-sen** [comment]
suborb Great fix!

Execute ex7 with blueMSX:
![ex7_0000](https://user-images.githubusercontent.com/5434159/66694916-4d9fbb80-ecf4-11e9-8756-b03461c09bea.png)
